server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -917,6 +914,22 @@ public String toString() {
         }
     }
 
+    // there is no equals on cluster state, so we just serialize it to XContent and compare Maps
+    // deserialized from the resulting JSON
+    private boolean assertPreviousStateConsistency(ClusterChangedEvent event) {
```

Comment:
I was able to reproduce very rare failures on this assertion that were the result of the routing nodes being ordered differently across the `previousState` and the `getLastAcceptedState` state.
With the serialization round trip and using the `equals` from the resulting map it seems completely stable.